rcl/src/rcl/arguments.c

```diff
@@ -1596,17 +1634,58 @@ rcl_ret_t
 _rcl_parse_log_level(
   const char * arg,
   rcl_allocator_t allocator,
-  int * log_level)
+  rcl_log_level_t * log_level)
 {
   RCL_CHECK_ARGUMENT_FOR_NULL(arg, RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(log_level, RCL_RET_INVALID_ARGUMENT);
 
-  rcutils_ret_t ret = rcutils_logging_severity_level_from_string(arg, allocator, log_level);
-  if (RCUTILS_RET_OK == ret) {
-    return RCL_RET_OK;
-  }
-  RCL_SET_ERROR_MSG("Argument does not use a valid severity level");
-  return RCL_RET_ERROR;
+  size_t comma_len = strlen(RCL_LOG_LEVEL_FLAG_COMMA);
+  size_t seperator_len = strlen(RCL_LOG_LEVEL_FLAG_SEPERATOR);
+  rcutils_ret_t ret = RCUTILS_RET_OK;
+  const char * p = arg;
+
+  do {
+    char * comma = strstr(p, RCL_LOG_LEVEL_FLAG_COMMA);
+    char * item;
+    if (comma == NULL) {
+      item = rcutils_strdup(p, allocator);
+    } else {
+      item = rcutils_strndup(p, comma - p, allocator);
+    }
```

Comment:
There's some error cases here that may not be covered.  I'm wondering about things like the following:

* `--log-level rcl=debug,`
* `--log-level rcl=debug,,`
* `--log-level rcl=`
* `--log-level rcl=,`

(there may be more, these are just some of the ones that came to mind)

Can you write tests for these cases to make sure we properly error and don't leak memory?